AssertionError: Tried to stop a LoopingCall that was not running.

Log:
```
...
2019-10-21 23:14:17,223 [twisted] CRITICAL:
Traceback (most recent call last):
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
    result = f(*args, **kw)
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/txmongo/connection.py", line 424, in __discovery_nodes_to_ping
    del self.__pingers[unused_peer]
KeyError: IPv4Address(type='TCP', host='XXX.XXX.XXX.XXX', port=27017)
...
...
2019-10-21 23:16:31,743 [scrapy.core.engine] INFO: Closing spider (finished)
2019-10-21 23:16:31,743 [scrapy.utils.signal] ERROR: Error caught on signal handler: <bound method MongoExtension.spider_closed of <amon.mongo_ext.MongoExtension object at 0x7fac7b2ff208>>
Traceback (most recent call last):
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
    result = f(*args, **kw)
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/pydispatch/robustapply.py", line 55, in robustApply
    return receiver(*arguments, **named)
  File "/opt/amon/amon/amon/mongo_ext.py", line 102, in spider_closed
    return disconnect()
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/txmongo/connection.py", line 346, in disconnect
    self.__pinger_discovery.stop()
  File "/opt/pyenv/versions/3.6.9/envs/amon/lib/python3.6/site-packages/twisted/internet/task.py", line 202, in stop
    assert self.running, ("Tried to stop a LoopingCall that was "
AssertionError: Tried to stop a LoopingCall that was not running.
...
```
I hide real host.

--- a/txmongo/connection.py
+++ b/txmongo/connection.py
@@ -343,7 +343,8 @@ class ConnectionPool(object):
         return db.command("dropDatabase")
 
     def disconnect(self):
-        self.__pinger_discovery.stop()
+        if self.__pinger_discovery.running:
+            self.__pinger_discovery.stop()
         for pinger in list(self.__pingers.values()):
             pinger.connector.disconnect()
 

